File: MiniGame-Java/src/main/java/rs/lab/mges/game/TitleScene.java
package rs.lab.mges.game;

import io.github.libsdl4j.api.render.SDL_Renderer;
import rs.lab.mges.engine.Control;
import rs.lab.mges.engine.SDLUtils;
import rs.lab.mges.engine.Text;
import rs.lab.mges.game.Mob.MobBee;
import rs.lab.mges.game.Mob.MobBlue;
import rs.lab.mges.game.Mob.MobFoxy;
import rs.lab.mges.game.Mob.MobSniky;
import rs.lab.mges.game.Powerup.Bomb;
import rs.lab.mges.game.Powerup.Medkit;
import rs.lab.mges.engine.SDLUtils.Vector2f;
import rs.lab.mges.engine.Drawable.AnimatedSprite;
import rs.lab.mges.game.RabbitGame.GameScene;

import static io.github.libsdl4j.api.render.SdlRender.SDL_RenderCopy;
import static io.github.libsdl4j.api.render.SdlRender.SDL_RenderFillRect;
import static io.github.libsdl4j.api.render.SdlRender.SDL_SetRenderDrawColor;

public class TitleScene extends Scene {

    private final MobBee bee;
    private final MobBlue blue;
    private final MobSniky sniky;
    private final MobFoxy foxy;
    private final Medkit medkit;
    private final Bomb bomb;
    private final HeroB heroB;

    public TitleScene(RabbitGame game) {
        super(game);

        bee = new MobBee(game, false).setPosition(new Vector2f(80.0f, 140.0f)).setVelocity(new Vector2f(0f, 0f));
        blue = new MobBlue(game, false).setPosition(new Vector2f(80.0f, 180.0f)).setVelocity(new Vector2f(0f, 0f));
        sniky = new MobSniky(game, false).setPosition(new Vector2f(80.0f, 220.0f)).setVelocity(new Vector2f(0f, 0f));
        foxy = new MobFoxy(game, false).setPosition(new Vector2f(80.0f, 260.0f)).setVelocity(new Vector2f(0f, 0f));
        bomb = new Bomb(game).setPosition(new Vector2f(80.0f, 300.0f)).setVelocity(new Vector2f(0f, 0f));
        medkit = new Medkit(game).setPosition(new Vector2f(80.0f, 340.0f)).setVelocity(new Vector2f(0f, 0f));
        heroB = new HeroB(game).setPosition(new Vector2f(40.0f, 40.0f)).setVelocity(new Vector2f(0f, 0f));
    }

    @Override
    void update(float dt) {
        if (Control.Fire) {
            game.changeScene(GameScene.GAMEPLAY);
            return;
        }

        bee.update(dt);
        blue.update(dt);
        sniky.update(dt);
        foxy.update(dt);

        heroB.update(dt);
    }

    @Override
    void draw(SDL_Renderer renderer, float delta) {
        SDLUtils.CheckSDLErr(() -> SDL_SetRenderDrawColor(renderer, Colors.DkGray2Color.r, Colors.DkGray2Color.g, Colors.DkGray2Color.b, Colors.DkGray2Color.a));
        SDLUtils.CheckSDLErr(() -> SDL_RenderFillRect(renderer, null));

        // static background
        var r = SDLUtils.rect(0, 16, game.w, 312);
        SDLUtils.CheckSDLErr(() -> SDL_RenderCopy(renderer, game.assets.forest, null, r));

        // title
        SDLUtils.CheckSDLErr(() -> SDL_RenderCopy(renderer, game.assets.title, null, null));

        heroB.draw(renderer, delta);

        // mobs
        bee.draw(renderer, delta);
        Text.drawText(renderer, 120, 150, Colors.WhiteColor, Colors.BlackColor, String.format("BEE .......... %3d points", bee.point));

        blue.draw(renderer, delta);
        Text.drawText(renderer, 120, 190, Colors.WhiteColor, Colors.BlackColor, String.format("BLUE ......... %3d points", blue.point));

        sniky.draw(renderer, delta);
        Text.drawText(renderer, 120, 230, Colors.WhiteColor, Colors.BlackColor, String.format("SNIKY ........ %3d points", sniky.point));

        foxy.draw(renderer, delta);
        Text.drawText(renderer, 120, 270, Colors.WhiteColor, Colors.BlackColor, String.format("FOXY ......... %3d points", foxy.point));

        bomb.draw(renderer, delta);
        Text.drawText(renderer, 120, 310, Colors.WhiteColor, Colors.BlackColor, String.format("BOMB ......... %3d points", bomb.point));

        medkit.draw(renderer, delta);
        Text.drawText(renderer, 120, 350, Colors.WhiteColor, Colors.BlackColor, String.format("MEDKIT........ %3d points", medkit.point));

        // texts
        Text.drawText(renderer, 140, 420, Colors.WhiteColor, Colors.DkGrayColor, "PRESS \"FIRE\" TO START");
        Text.drawText(renderer, 135, 480 - 18, Colors.DkGrayColor, Colors.BlackColor, "PRESS \"F12\" TO QUIT OR \"F11\" FOR FULLSCREEN",
                Text.DrawTextOptions.builder().setSize(0.5f));
    }

    @Override
    void enter() {
        Control.reset();
    }

    private static class HeroB extends AnimatedSprite {

        public HeroB(RabbitGame game) {
            super(game.assets.heroBFrames, 64, 64);
        }

        @Override
        public void update(float dt) {
            animTimer += dt;
            if (animTimer > 0.3f) {
                animTimer = 0.0f;
                animFrame = 1 - animFrame;
            }
        }
    }
}

File: MiniGame-Python/src/game/TitleScene.py
from .Scene import Scene, GameScene
from . import Colors
from .Mob import MobBee, MobBlue, MobFoxy, MobSniky
from .Powerup import Bomb, Medkit

from engine import Control, Text, Vector2f, AnimatedSprite, DrawTextOptions
from sdl2 import *  # type: ignore


class TitleScene(Scene):
    def __init__(self, game):
        super().__init__(game)

        self._bee = MobBee(self._game, False)
        self._bee.position = Vector2f(80.0, 140.0)
        self._bee.velocity = Vector2f(0.0, 0.0)

        self._blue = MobBlue(self._game, False)
        self._blue.position = Vector2f(80.0, 180.0)
        self._blue.velocity = Vector2f(0.0, 0.0)

        self._sniky = MobSniky(self._game, False)
        self._sniky.position = Vector2f(80.0, 220.0)
        self._sniky.velocity = Vector2f(0.0, 0.0)

        self._foxy = MobFoxy(self._game, False)
        self._foxy.position = Vector2f(80.0, 260.0)
        self._foxy.velocity = Vector2f(0.0, 0.0)

        self._bomb = Bomb(self._game)
        self._bomb.position = Vector2f(80.0, 300.0)
        self._bomb.velocity = Vector2f(0.0, 0.0)

        self._medkit = Medkit(self._game)
        self._medkit.position = Vector2f(80.0, 340.0)
        self._medkit.velocity = Vector2f(0.0, 0.0)

        self._heroB = HeroB(self._game)
        self._heroB.position = Vector2f(40.0, 40.0)
        self._heroB.velocity = Vector2f(0.0, 0.0)


    def enter(self):
        Control.reset()

    def update(self, dt):
        if Control.Fire:
            self._game.changeScene(GameScene.GAMEPLAY)
            return

        self._bee.update(dt)
        self._blue.update(dt)
        self._sniky.update(dt)
        self._foxy.update(dt)

        self._heroB.update(dt)

    def draw(self, renderer, delta):
        SDL_SetRenderDrawColor(
            renderer,
            Colors.DkGray2Color.r,
            Colors.DkGray2Color.g,
            Colors.DkGray2Color.b,
            0xff,
        )

        SDL_RenderFillRect(renderer, None)

        # Static background (forest)
        r = SDL_Rect(0, 16, self._game._w, 312)
        SDL_RenderCopy(renderer, self._game.assets.forest, None, r)

        # Title
        SDL_RenderCopy(renderer, self._game.assets.title, None, None)
        self._heroB.draw(renderer, delta)

        # Mobs
        self._bee.draw(renderer, delta)
        Text.drawText(renderer, 120, 150, Colors.WhiteColor, Colors.BlackColor, f"BEE .......... {self._bee._points:3} points")

        self._blue.draw(renderer, delta)
        Text.drawText(renderer, 120, 190, Colors.WhiteColor, Colors.BlackColor, f"BLUE ......... {self._blue._points:3} points")

        self._sniky.draw(renderer, delta)
        Text.drawText(renderer, 120, 230, Colors.WhiteColor, Colors.BlackColor, f"SNIKY ........ {self._sniky._points:3} points")

        self._foxy.draw(renderer, delta)
        Text.drawText(renderer, 120, 270, Colors.WhiteColor, Colors.BlackColor, f"FOXY ......... {self._foxy._points:3} points")

        self._bomb.draw(renderer, delta)
        Text.drawText(renderer, 120, 310, Colors.WhiteColor, Colors.BlackColor, f"BOMB ......... {self._bomb._points:3} points")

        self._medkit.draw(renderer, delta)
        Text.drawText(renderer, 120, 350, Colors.WhiteColor, Colors.BlackColor, f"MEDKIT ....... {self._medkit._points:3} points")


        # Additional texts
        Text.drawText(renderer, 140, 420, Colors.WhiteColor, Colors.DkGrayColor, "PRESS \"FIRE\" TO START")
        Text.drawText(renderer, 135, 480 - 18, Colors.DkGrayColor, Colors.BlackColor, "PRESS \"F12\" TO QUIT OR \"F11\" FOR FULLSCREEN", 
                      DrawTextOptions(size=0.5))

class HeroB(AnimatedSprite):
    def __init__(self, game):
        super().__init__(game.assets.heroBFrames, 64, 64)

    def update(self, dt):
        self._animTimer += dt
        if self._animTimer > 0.3:
            self._animTimer = 0.0
            self._animFrame = 1 - self._animFrame
